// Portions of the code are licensed under the BSD:
// MeCab -- Yet Another Part-of-Speech and Morphological Analyzer
//
//
//  Copyright(C) 2001-2006 Taku Kudo <taku@chasen.org>
//  Copyright(C) 2004-2006 Nippon Telegraph and Telephone Corporation
// All rights reserved.
//
// Redistribution and use in source and binary forms, with or without modification, are
// permitted provided that the following conditions are met:
//
//  * Redistributions of source code must retain the above
//     copyright notice, this list of conditions and the
//        following disclaimer.
//
//  * Redistributions in binary form must reproduce the above
//     copyright notice, this list of conditions and the
//        following disclaimer in the documentation and/or other
//           materials provided with the distribution.
//
//  * Neither the name of the Nippon Telegraph and Telegraph Corporation
//     nor the names of its contributors may be used to endorse or
//        promote products derived from this software without specific
//           prior written permission.
//
// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED
// WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
// PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
// ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR
// TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

// Relicensed under the Artistic License 2.0:
// raku-MeCab -- A Raku bindings for MeCab
//  Copyright(C) 2016 titsuki <titsuki@cpan.org>
// For more details, see the "LICENSE" file.
--- mecab-0.996/src/mecab.h	2013-02-18 02:21:55.000000000 +0900
+++ mecab-0.996-new/src/mecab.h	2016-10-25 00:33:30.941979127 +0900
@@ -753,6 +753,9 @@
                                                     const char *end,
                                                     mecab_lattice_t *lattice);
 
+  /* node interface */
+  MECAB_DLL_EXTERN char         *mecab_node_t_surface_get(mecab_node_t *node);
+
   /* static functions */
   MECAB_DLL_EXTERN int           mecab_do(int argc, char **argv);
   MECAB_DLL_EXTERN int           mecab_dict_index(int argc, char **argv);
